internal/pkg/peer/blocksprovider/util.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package blocksprovider
+
+import (
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/hyperledger/fabric/internal/pkg/peer/orderers"
+)
+
+const (
+	bftMinBackoffDelay           = 10 * time.Millisecond
+	bftMaxBackoffDelay           = 10 * time.Second
+	bftBlockRcvTotalBackoffDelay = 20 * time.Second
+	bftBlockCensorshipTimeout    = 20 * time.Second
+)
+
+type errRefreshEndpoint struct {
+	message string
+}
+
+func (e *errRefreshEndpoint) Error() string {
+	return e.message
+}
+
+type errStopping struct {
+	message string
+}
+
+func (e *errStopping) Error() string {
+	return e.message
+}
+
+type errFatal struct {
+	message string
+}
+
+func (e *errFatal) Error() string {
+	return e.message
+}
+
+func backOffDuration(base float64, exponent uint, minDur, maxDur time.Duration) time.Duration {
+	if base < 1.0 {
```

Comment:
base is generic but you set it to a constant where you call it.
Why not just pass the constant all the way here and avoid this check? 

and why use nano second resolution? Why not just use milliseconds resolution?

Reply:
The base may be different in different places. For example in the old CFT code it is 1.2 and In the new BFT code it is 2.0. 
This function is flexible enough to cope with different bases.
Nanos-seconds is the underlying resolution of time.Duration (int64 in ns). I see no reason the decrease the resolution.